### DISCUSSION TITLE
Required and Validate in rules are not validated in turn

### DISCUSSION THREAD

**Discussion by infothien** [react-hook-form/react-hook-form#2843] [Q&A] [answered]
I got the simple Field like this with 2 validation 
<img width="403" alt="RegisterProfile_tsx_—_dentsu" src="https://user-images.githubusercontent.com/30365498/92384716-8b4ad580-f14b-11ea-931e-e03b7159802e.png">

Note: the validation limitLength here is just example, definitely I got something else need to use validate property in rules.

Problem: I realized that, even my textField is empty but it will show the limitLength validation message instead of "it is required" message 
<img width="187" alt="Screenshot_2020_09_07_20_44" src="https://user-images.githubusercontent.com/30365498/92384478-1e374000-f14b-11ea-853e-25c55038c5bc.png">

Expected behavior:
Should just validate the limitLength if the current value is not empty and if the current value is empty should check required validation, As long as required validation pass, then it should check the limitLength

Thank you, hope to get some helps from you guys

**Comment by bluebill1049** [comment, marked as answer]
can you provide a CSB?

**Reply by infothien** [reply]
@bluebill1049 thank for your reply.
https://codesandbox.io/s/loving-worker-ri0g7?file=/src/App.js:0-801, that is super weird I got the same code in this sandbox for my real react-native app, in the mobile app it got the problem but when testing on codeSandbox, seem to me it is working correctly @@?

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/30365498/92390132-488dfb00-f155-11ea-83c7-aa70aaadd1f0.gif)

**Reply by infothien** [reply]
Sorry, i think I got problem b/c passing {...props} to the TextInput component, could solve it right now

**Reply by bluebill1049** [reply]
oh nice! glad you work it out.
